### ISSUE TITLE
media-query-no-invalid doesn't work with scss variables?

### ISSUE THREAD

**Issue by jdelStrother** [stylelint/stylelint#7071]
### What minimal example or steps are needed to reproduce the bug?

```css
$width: 350px;

@media screen and (min-width: $width) {
  .xxx {
    display: none;
  }
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": ["stylelint-scss"],
  "rules": {
    "media-query-no-invalid": true
  }
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIAkB3AllALgC3gAgGYBWABgAcAPAbgB0A7egAQFsYo0BDPAZzACcYMOng50oeABTM0dALTosuPKgw4AlHmD0amAHQUDm+nhN523MgBsOAT3x0IdGLWF4AvvQ90QAGhAAzNEsYADkOVgQQGApwqxhdXm5uX3BHQIBzSOZoAFdg3WiyCD5MbjwAXiNXKxz0mW58AG1jUxoQbkwbYMsZTFlE7jafFoBdYdc+PJgGqp0dAHJWdg5ZAEccmD4bWQdZGQA3Dh6oefxMSZgWrzcXFMg6DIAxYuYOTEiAK2S-WDJkxC0rjaHS6MB6dEwbXwbWsmGmkN8LWBnW6vVk9wy-UwoigHD4UCheBhb3hQyR7RRYLRGLQ6SxOLxUH6YCShOJcI6ZKBFNB4L6NLpLO4smyUCmgwQRJAsNJIE8IDcQA)

### Which Stylelint-related dependencies are you using?

```json
{
    "stylelint": "15.10.1",
    "stylelint-config-recommended": "13.0.0",
    "stylelint-config-standard-scss": "10.0.0",
    "stylelint-scss": "5.0.1"
}
```

### What did you expect to happen?

No problems to be reported

### What actually happened?

```
ERROR
Unexpected invalid media query "(min-width: $width)"
[(media-query-no-invalid)](https://stylelint.io/user-guide/rules/media-query-no-invalid)
[3:19-38]
```

**Comment by jeddy3** [comment]
@jdelStrother Thanks for the report and for using the template.

The rule correctly reports `(min-width: $width)` as invalid according to the CSS specifications.

From the [rule docs](https://stylelint.io/user-guide/rules/media-query-no-invalid/):

> Media queries must be grammatically valid according to the [Media Queries Level 5](https://www.w3.org/TR/mediaqueries-5/) specification.

> This rule is only appropriate for CSS. You should not turn it on for CSS-like languages, such as Sass or Less, as they have their own syntaxes.

Stylelint and the built-in rules are geared towards standard CSS. If you use a language extension like SCSS, you'll need to turn off the incompatible rules.

It looks like the authors of the [stylelint-config-recommended-scss](https://github.com/stylelint-scss/stylelint-config-recommended-scss) plugin [are already planning to do this](https://github.com/stylelint-scss/stylelint-config-recommended-scss/commit/c95076bbf5a2a1443872241fa961388d074a23e6). So, you can wait until their next release is out or turn off the rule yourself.

As an aside, you don't need to include `stylelint-config-recommended` nor `stylelint-scss` as these are bundled in `stylelint-config-standard-scss`:

```json
{
    "stylelint": "15.10.1",
    "stylelint-config-standard-scss": "10.0.0",
}
```

```
{
  "extends": ["stylelint-config-standard-scss"],
  "rules": {
    "media-query-no-invalid": null
  }
}
```

**Comment by jdelStrother** [comment]
> It looks like the authors of the [stylelint-config-recommended-scss](https://github.com/stylelint-scss/stylelint-config-recommended-scss) config [are already planning to do this](https://github.com/stylelint-scss/stylelint-config-recommended-scss/commit/c95076bbf5a2a1443872241fa961388d074a23e6). So, you can wait until their next release is out or turn off the rule yourself.

Ohhh, thanks. Sorry - I did search their issues and didn't find any mention of it, but I didn't check to see if it was already fixed in the code.
